### PR TITLE
Remove `rustc::existing_doc_keyword` lint

### PR DESCRIPTION
**Pull request by nnethercote**
The check doesn't require a lint.

r? @GuillaumeGomez

### CODE REVIEW

**Comment by nnethercote** [comment]
Details in individual commits.

File: tests/rustdoc-gui/src/test_docs/lib.rs [hunk 1]
```diff
@@ -156,7 +156,7 @@ pub enum AnEnum {
     WithVariants { and: usize, sub: usize, variants: usize },
 }
 
-#[doc(keyword = "CookieMonster")]
+#[doc(keyword = "auto")]
```
Comment (GuillaumeGomez) [inline tests/rustdoc-gui/src/test_docs/lib.rs:159]: Why this change?

File: tests/rustdoc-json/keyword.rs [hunk 2]
```diff
@@ -15,6 +15,6 @@ pub mod foo {}
 
 //@ !has "$.index[*][?(@.name=='hello')]"
 //@ !has "$.index[*][?(@.name=='bar')]"
-#[doc(keyword = "hello")]
+#[doc(keyword = "break")]
```
Comment (GuillaumeGomez) [inline tests/rustdoc-json/keyword.rs:18]: Why this change?

File: tests/ui/internal-lints/existing_doc_keyword.rs [hunk 3]
```diff
@@ -1,10 +0,0 @@
-//@ compile-flags: -Z unstable-options
```
Comment (GuillaumeGomez) [inline tests/ui/internal-lints/existing_doc_keyword.rs:1]: Why removing this test?

Comment (GuillaumeGomez) [reply tests/ui/internal-lints/existing_doc_keyword.rs:1]: Ah I see, nevermind. :)

File: tests/rustdoc-gui/src/test_docs/lib.rs (hunk 1 above)
Comment (nnethercote) [reply tests/rustdoc-gui/src/test_docs/lib.rs:159]: The test now fails if the value isn't a valid keyword.

Comment (GuillaumeGomez) [reply tests/rustdoc-gui/src/test_docs/lib.rs:159]: Well, that brings this question: why this change?

Comment (nnethercote) [reply tests/rustdoc-gui/src/test_docs/lib.rs:159]: Previously the "is it a valid keyword?" test was done by the lint. It has now moved to `check_doc_keyword` and it runs in several test cases where the lint did not.

Comment (GuillaumeGomez) [reply tests/rustdoc-gui/src/test_docs/lib.rs:159]: No I mean, `doc(keyword)` was voluntarily left "open" in what kind of keyword we could define. You changed this behaviour. That's the part I don't understand. I don't mind changing it away from being a lint.

Comment (nnethercote) [reply tests/rustdoc-gui/src/test_docs/lib.rs:159]: The lint checked this:
```rust
fn is_doc_keyword(s: Symbol) -> bool {
    s <= kw::Union
}
```
That hasn't changed (well, `sym::SelfTy` is now also accepted, for the exception). What has changed is how often this check is being applied.

Comment (GuillaumeGomez) [reply tests/rustdoc-gui/src/test_docs/lib.rs:159]: I see, my bad.

**Comment by nnethercote** [comment]
@GuillaumeGomez: the `tests/rustdoc-gui/search-result-color.goml` is driving me crazy. I can't understand how it works. The keyword used for it is `CookieMonster`. The docs for the keyword are given in `tests/rustdoc-gui/src/test_docs/lib.rs` and the test search is specified by `tests/rustdoc-gui/search-result-keyword.goml`.

It seems like I should just be able to change `CookieMonster` in those two files to any other identifier. In practice, some changes work and some don't. Some examples:

- Run the test (`x test tests/rustdoc-gui/search-result-color.goml)`: pass

- Change `CookieMonster` to `ZookieMonster` in
  `tests/rustdoc-gui/src/test_docs/lib.rs` and
  `tests/rustdoc-gui/search-result-keyword.goml`, run test: fail (3 errors)

- Change `ZookieMonster` back to `CookieMonster`, run test: fail (4 errors)

- Delete `build/x86_64-unknown-linux-gnu/test/`, run test: pass

- Change `CookieMonster` to `CookieThief`, run test: pass

- Change `CookieThief` to `Mookie`, run test: fail (3 errors)

- Delete `build/x86_64-unknown-linux-gnu/test/`, run test: fail (3 errors)

- Change `Mookie` to `Cook`, run test: fail (4 errors)

- Delete `build/x86_64-unknown-linux-gnu/test/`, run test: pass

Keywords that are similar to `CookieMonster` and start with `C` tend to work. Those that aren't similar fail. Sometimes deleting the `test/` dir is required to get things working again after a failed attempt.

I have viewed the generated docs in a browser. They always work fine no matter what keyword I use, and the search always works as expected. So I'm baffled. I just want to change `CookieMonster` to any existing keyword (such as `auto`) to satisfy the keyword check. Any ideas what is happening with this test?

File: tests/rustdoc-gui/search-result-keyword.goml [hunk 4]
```diff
@@ -1,8 +1,8 @@
 // Checks that the "keyword" results have the expected text alongside them.
 go-to: "file://" + |DOC_PATH| + "/test_docs/index.html"
-write-into: (".search-input", "CookieMonster")
+write-into: (".search-input", "auto")
 // To be SURE that the search will be run.
 press-key: 'Enter'
 // Waiting for the search results to appear...
 wait-for: "#search-tabs"
-assert-text: (".result-keyword .result-name", "keyword CookieMonster")
+assert-text: (".result-keyword .result-name", "keyword auto")
```
[the remark this answers is not in the document]
Comment (nnethercote) [reply tests/rustdoc-gui/search-result-keyword.goml:8]: I checked for that, `auto` is the first item. `ZookieMonster` was too, as were a bunch of other things I tried.

File: tests/rustdoc-gui/search-result-keyword.goml [hunk 5]
```diff
@@ -1,8 +1,8 @@
 // Checks that the "keyword" results have the expected text alongside them.
 go-to: "file://" + |DOC_PATH| + "/test_docs/index.html"
-write-into: (".search-input", "CookieMonster")
+write-into: (".search-input", "auto")
```
[the remark this answers is not in the document]
Comment (nnethercote) [reply tests/rustdoc-gui/search-result-keyword.goml:3]: I'll try the keyword change on Monday. I did already try testing manually in a browser, it always worked fine no matter what I tried. That's why it was so baffling why some things worked but most didn't -- I could find an explanation to explain the success/failure differences.

**Comment by GuillaumeGomez** [comment]
Found the issue in `search-result-color.goml`: there is no keyword displayed, meaning the test will fail since we can't check the color for a keyword. We need to use another keyword that might match other items.

**Comment by GuillaumeGomez** [comment]
I pushed a fix to your branch. To keep the `fo` query working, I used the `for` keyword instead.

**Comment by nnethercote** [comment]
Thank you! It was the `"/test_docs/index.html?search=coo"` that I overlooked. That explains why `CookieMonster` and `CookieThief` and other variations like that worked, but other terms didn't.

**Comment by nnethercote** [comment]
I'm happy for this to merge if you are.

**Comment by GuillaumeGomez** [comment]
I am too so let's go!

@bors r=GuillaumeGomez,nnethercote

File: tests/rustdoc-json/keyword.rs [hunk 6]
```diff
@@ -15,6 +15,6 @@ pub mod foo {}
 
 //@ !has "$.index[*][?(@.name=='hello')]"
```
Comment (aDotInTheVoid) [inline tests/rustdoc-json/keyword.rs:16]: This should be changed to check “break” instead of “hello”.

**Review by aDotInTheVoid — COMMENTED** [review]
@bors r- until rustdoc-json text fixed.

File: tests/rustdoc-json/keyword.rs (hunk 6 above)
Comment (nnethercote) [reply tests/rustdoc-json/keyword.rs]: ok
